pkg/cmd/serve.go

```diff
@@ -45,121 +49,149 @@ var (
 	}
 
+	experimentalFlags := nfs.FlagSet(BoldBlue("Experimental"))
+	// Flags for experimental features
+	experimentalFlags.BoolVar(&config.EnableExperimentalLookupResources, "enable-experimental-lookup-resources", false, "enables the experimental version of the lookup resources API")
+	experimentalFlags.BoolVar(&config.EnableExperimentalWatchableSchemaCache, "enable-experimental-watchable-schema-cache", false, "enables the experimental schema cache which makes use of the Watch API for automatic updates")
+	// TODO: these two could reasonably be put in either the Dispatch group or the Experimental group. Is there a preference?
+	experimentalFlags.StringToStringVar(&config.DispatchSecondaryUpstreamAddrs, "experimental-dispatch-secondary-upstream-addrs", nil, "secondary upstream addresses for dispatches, each with a name")
+	experimentalFlags.StringToStringVar(&config.DispatchSecondaryUpstreamExprs, "experimental-dispatch-secondary-upstream-exprs", nil, "map from request type (currently supported: `check`) to its associated CEL expression, which returns the secondary upstream(s) to be used for the request")
+
+	observabilityFlags := nfs.FlagSet(BoldBlue("Observability"))
+	// Flags for observability and profiling
+	otel := cobraotel.New(cmd.Use)
+	otel.RegisterFlags(observabilityFlags)
+	runtime.RegisterFlags(observabilityFlags)
+
+	metricsFlags := nfs.FlagSet(BoldBlue("Metrics Server"))
+	// Flags for metrics
+	util.RegisterHTTPServerFlags(metricsFlags, &config.MetricsAPI, "metrics", "metrics", ":9090", true)
+
+	telemetryFlags := nfs.FlagSet(BoldBlue("Telemetry"))
+	// Flags for telemetry
+	telemetryFlags.StringVar(&config.TelemetryEndpoint, "telemetry-endpoint", telemetry.DefaultEndpoint, "endpoint to which telemetry is reported, empty string to disable")
+	telemetryFlags.StringVar(&config.TelemetryCAOverridePath, "telemetry-ca-override-path", "", "TODO")
+	telemetryFlags.DurationVar(&config.TelemetryInterval, "telemetry-interval", telemetry.DefaultInterval, "approximate period between telemetry reports, minimum 1 minute")
+
+	miscellaneousFlags := nfs.FlagSet(BoldBlue("Miscellaneous"))
+	// Flags for things that don't neatly fit into another bucket
+	termination.RegisterFlags(miscellaneousFlags)
+
 	// Flags for misc services
-	util.RegisterHTTPServerFlags(cmd.Flags(), &config.MetricsAPI, "metrics", "metrics", ":9090", true)
 
 	if err := util.RegisterDeprecatedHTTPServerFlags(cmd, "dashboard", "dashboard"); err != nil {
 		return err
 	}
 
-	// Flags for telemetry
-	cmd.Flags().StringVar(&config.TelemetryEndpoint, "telemetry-endpoint", telemetry.DefaultEndpoint, "endpoint to which telemetry is reported, empty string to disable")
-	cmd.Flags().StringVar(&config.TelemetryCAOverridePath, "telemetry-ca-override-path", "", "TODO")
-	cmd.Flags().DurationVar(&config.TelemetryInterval, "telemetry-interval", telemetry.DefaultInterval, "approximate period between telemetry reports, minimum 1 minute")
-
```

Comment:
You need to add this to the end in order for the flags to actually be registered with the command. This seems like an unintuitive DX for `cobrautil`:
```go
nfs.AddFlagSets(cmd)
```